client/signup/steps/design-picker/index.jsx

```diff
@@ -285,13 +318,13 @@ export default function DesignPickerStep( props ) {
 				align={ isMobile ? 'left' : 'center' }
 				hideSkip
 				hideNext={ false }
-				nextLabelText={ translate( 'Start with %(designTitle)s', {
-					args: { designTitle },
-				} ) }
+				nextLabelText={ nextLabelText }
 				defaultDependencies={ defaultDependencies }
 				backUrl={ getStepUrl( flowName, stepName, '', locale, queryParams ) }
-				goToNextStep={ submitDesign }
+				goToNextStep={ shouldUpgrade ? upgradePlan : submitDesign }
 				stepSectionName={ designTitle }
+				// Prevent from completing the flow when the user clicks on Upgrade Plan button
+				nextDisabledSubmitOnClick={ shouldUpgrade }
```

Comment:
So, I was reading [Modular Signup Framework README](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/README.md), and I feel that if we introduce this "override", it would complicate the continuity of the flow: now, the "next" button does not always move us forward in the signup progress, it needs to look for this props.

Instead, what if we introduce another button type in `StepWrapper`, say, an "action button", which explicitly does not alter the progress? It just needs two props: `actionLabelText` and `actionOnClick` which can be populated with our upgrade flow.

I've implemented this rough idea here: https://github.com/Automattic/wp-calypso/commit/2d2249cc275ebc04f99d0f08d05f7dbcaad0c860. This seems to work locally.

Feel free to consider! Not sure if it's too late already to refactor this. I implemented the above mainly for my codebase exploratory purposes 😄

Reply:
I think it's a good idea. I don't do this because there are lots of buttons there 😅 But maybe this is a better choice. However, I think we can introduce `actionButtons` or `customizedActionButtons` and leave the steps to decide what button they want to provide more flexibility.

Reply:
> and leave the steps to decide what button they want to provide more flexibility.

We can do that, but then the clients will need to style the buttons, whereas `<StepWrapper />` already has the CSS (rounded buttons). That's why I put the button there instead. However will leave the implementation details to you 🙈

Reply:
Yep, I know but I'm afraid the button becomes more and more here 🥲